### PR TITLE
feat: add missing list expr namespace methods and tweaks

### PR DESCRIPTION
**Pull request by etiennebacher**

### CODE REVIEW

File: R/expr-list.R [hunk 1]
```diff
@@ -460,9 +475,10 @@ expr_list_head <- function(n = 5L) {
 #'   tail_by_lit = pl$col("s")$list$tail(2)
 #' )
 expr_list_tail <- function(n = 5L) {
-  offset <- -as_polars_expr(n)
-  self$`_rexpr`$list_slice(offset$`_rexpr`, as_polars_expr(n)$`_rexpr`) |>
-    wrap()
+  wrap({
+    offset <- -as_polars_expr(n)
```
Comment (eitsupi) [inline R/expr-list.R:479]: Please use the `check_number_decimal` function to make sure that n is numeric value before that.
Like this:
https://github.com/eitsupi/neo-r-polars/blob/ec7b1e9c8ae609a14bc3ef0ab819c5254e034aee/R/dataframe-frame.R#L178

**Review by eitsupi — CHANGES_REQUESTED** [review]

File: R/expr-list.R (hunk 1 above)
Comment (etiennebacher) [reply R/expr-list.R:479]: But `n` can be a string (see in the examples above) so this check cannot be done.

Comment (eitsupi) [reply R/expr-list.R:479]: Thanks, you are right. I was mistaken.

**Review by eitsupi — APPROVED** [review]
Thanks for checking the details. Looks good, so let's merge it.

**Pull request merged**